### ISSUE TITLE
Zero-shot on Libero

### ISSUE THREAD

**Issue by xianjiedai** [Physical-Intelligence/openpi#415]
Hi,

Thanks for the great work!

I am wondering whether you have conducted any zero-shot experiments on Libero dataset. I use the **pi0_libero** config and **pi0_base** as the checkpoint. However, I got 0 Suceess Rate while running the Libero simulation experiment.

It seems I configed everything with mistakes. I am wondering whether it's normal to have 0 SR?

Best

**Comment by kpertsch** [comment]
Hi!

The pi0_base and pi0_fast_base checkpoints have **not** been trained with any Libero data in the mix (no sim data at all for that matter). So we don't expect them to work on the Libero environments out of the box. We included Libero as an example for how these models can be adapted to a new robot platform with a bit of in-domain fine-tuning data, which is what resulted in the pi0_libero and pi0_fast_libero checkpoints. Those should work "out of the box" on the Libero evals.

Hope this clarifies things! Will close this issue for now, but feel free to reopen!

**Comment by jaelyontway** [comment]
Hi, 

How do you do the zeroshot with the pi0 model, not a pi0-FAST model? I encountered an error of normalization. 
I used dir: pi0_base in serve_policy.py and ran the same command as provided in examples/libero/readme

**Comment by githubLeoliu** [comment]
Hi, 

I was trying conduct zero-shot experiment on Libero with **pi0_fast_libero** config and **pi0_fast_base** checkpoint. The reason I got 0 SR seems like cause by shape mismatch between the model and the environment. 

<img width="1236" height="188" alt="Image" src="https://github.com/user-attachments/assets/556c27e9-652c-4bea-ab62-99fac2a02578" />

I suppose this is because the tokenizer was not trained on Libero. I also notices that there is a universal tokenizer called FAST+ mentioned in the paper. Does FAST+ sill need finetuning to be able to work on Libero?

**Comment by kpertsch** [comment]
This is not an issue of the tokenizer but the model -- this is expected for the pi0_fast_base model, since as mentioned above it is not trained on Libero.
For pi0_fast_libero I would assume that decoding errors would be very rare and the success rate should be significantly higher than 0, so that suggests something about your setup is incorrect.
